code/target/src/main/java/com/adobe/marketing/mobile/target/TargetPreviewManager.java

```diff
@@ -362,20 +380,32 @@ private boolean setupTargetPreviewParameters(final Map<String, String> previewPa
      *
      */
     private void createAndShowFloatingButton() {
-        if (floatingButton != null) {
+        if (floatingButtonPresentable != null) {
             Log.debug(TargetConstants.LOG_TAG, CLASS_NAME, "createAndShowFloatingButton - Floating button already exists");
             return;
         }
 
-        TargetPreviewButtonListener targetPreviewButtonListener = new TargetPreviewButtonListener(this);
-        floatingButton = uiService.createFloatingButton(targetPreviewButtonListener);
+        if (context.getAssets() == null) {
```

Comment:
nit: Store this in a var so that you do not need to reinvoke the method (L394) upon non-null return value.